### PR TITLE
feat(28): adding testing packages and sendEmail for confirmation

### PR DESCRIPTION
**Pull request by 404answernotfound**
### 🔗 Linked issue

#28 

- Added a new `/emails` page where you will find a button to send a confirmation email.
- Sends confirmation email once subscription process is completed (right now not working because of Stripe probably)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/app/actions/createMembership.tsx [hunk 1]
```diff
@@ -118,6 +121,17 @@ export async function createMembership(
     },
   })
 
+  // Send confirmation email
+  const confirmationEmail = render(<MembershipConfirmationEmail />)
+
+  await mailer().sendMail({
+    from: "noreply@schrodinger-hat.org",
```
Comment (TheJoin95) [inline src/app/actions/createMembership.tsx:128]: Wrong email domain

File: src/app/actions/createMembership.tsx [hunk 2]
```diff
@@ -118,6 +121,17 @@ export async function createMembership(
     },
   })
 
+  // Send confirmation email
+  const confirmationEmail = render(<MembershipConfirmationEmail />)
```
Comment (TheJoin95) [inline src/app/actions/createMembership.tsx:125]: Do we really want to render here? Could we use other methods to get the email html?

File: src/emails/membershipConfirmedEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            <Text style={footer}>{"MARKETING_EMAIL_NAME"}</Text>
+            <Text style={footer}>{"MARKETING_EMAIL_ADDRESS"}</Text>
+          </Container>
+        </Body>
+      </Html>
+    </>
+  )
+}
+
+const main = {
```
Comment (TheJoin95) [inline src/emails/membershipConfirmedEmail.tsx:63]: Could we a common style to reuse on other email flow?

File: src/app/actions/createMembership.tsx (hunk 2 above)
Comment (404answernotfound) [reply src/app/actions/createMembership.tsx:125]: This is how the documentation is showing it and it seems to be good enough. I guess we could potentially render it beforehand and export it. Worth a try?

File: src/emails/membershipConfirmedEmail.tsx (hunk 3 above)
Comment (404answernotfound) [reply src/emails/membershipConfirmedEmail.tsx:63]: We could, but for now it's not worth it. FIrst thing is to ensure this actually works!

File: src/app/actions/createMembership.tsx (hunk 2 above)
Comment (TheJoin95) [reply src/app/actions/createMembership.tsx:125]: Alright, yeah, maybe an export before hand would be great I believe

File: src/emails/membershipConfirmedEmail.tsx (hunk 3 above)
Comment (TheJoin95) [reply src/emails/membershipConfirmedEmail.tsx:63]: Oki doki

**Comment by 404answernotfound** [comment]
I'm not actually _receiving_ emails, but we are sending them.

Screenshot 2024-10-16 at 20 33 13

File: src/app/actions/createMembership.tsx (hunk 1 above)
Comment (TheJoin95) [reply src/app/actions/createMembership.tsx:125]: @404answernotfound please look at this

**Comment by TheJoin95** [comment]
Please rebase

File: src/app/api/send-email/route.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+// src/app/api/send-email/route.ts
+import { NextResponse } from "next/server"
+import { mailer } from "@/services/nodemailer"
+import { confirmationEmail } from "@/emails/renders/membershipConfirmedEmailRender"
+
+export async function POST() {
+  try {
+    const response = await mailer().sendMail({
+      from: "noreply@schrodinger-hat.org", // TODO: needs to be updated
```
Comment (TheJoin95) [inline src/app/api/send-email/route.ts:9]:
```suggestion
      from: "noreply@schroedinger-hat.org", // TODO: needs to be updated
```

**Comment by 404answernotfound** [comment]
@TheJoin95 rebased

**Comment by TheJoin95** [comment]
@404answernotfound please check the CI

File: src/app/actions/createMembership.tsx [hunk 5]
```diff
@@ -118,6 +120,15 @@ export async function createMembership(
     },
   })
 
+  // Send confirmation email
+  await mailer().sendMail({
+    from: "noreply@schrodinger-hat.org",
```
Comment (TheJoin95) [inline src/app/actions/createMembership.tsx:125]: Could we have it as an ENV variable?

File: src/app/api/send-email/route.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// src/app/api/send-email/route.ts
+import { NextResponse } from "next/server"
+import { mailer } from "@/services/nodemailer"
+import { confirmationEmail } from "@/emails/renders/membershipConfirmedEmailRender"
+
+export async function POST() {
+  try {
+    const response = await mailer().sendMail({
+      from: "noreply@schroedinger-hat.org", // TODO: needs to be updated
```
Comment (TheJoin95) [inline src/app/api/send-email/route.ts:9]: Same as previous comment about .env

File: src/app/emails/layout.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        <div className="relative hidden h-full flex-col bg-muted p-10 text-white dark:border-r lg:flex">
+          <div className="absolute inset-0 bg-zinc-900" />
+          <div className="relative z-20 flex items-center text-lg font-medium">
+            <Image src={logoWhite} alt={"logo"} width={42} className={"mr-3"} />
+            {"MARKETING_NAME"}
+          </div>
+          <div className="relative z-20 mt-auto">
+            <blockquote className="space-y-2">
+              <p className="text-lg">
+                &ldquo;Un socio é un pó come un amico.&rdquo;
```
Comment (TheJoin95) [inline src/app/emails/layout.tsx:32]: Weird accent here

**Comment by TheJoin95** [comment]
Overall looking good, I think we can merge it after the previous comments are going to be resolved, thanks!

**Comment by 404answernotfound** [comment]
Using this PR to also get a start on #29 and #30 which will require admin membership page to be completed before moving forward. This is enough to get us started with making emails work tho, so it's a closing for #28 and a starting point for debugging #29 and #30

To your points @TheJoin95 , we can add env vars

File: src/app/emails/layout.tsx (hunk 7 above)
Comment (404answernotfound) [reply src/app/emails/layout.tsx:32]: This is taken from previous UI state so let's get back to it at a later point if needs changing

**Review by TheJoin95 — APPROVED** [review]
Approving so that after fixing the lint issues you can merge it

**Comment by 404answernotfound** [comment]
Should have fixed `linting issue` that didn't come with this PR but we can still fix it here

**Pull request merged**